### Importación de datos



In [2]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

print(tienda.shape)
print(tienda2.shape)
print(tienda3.shape)
print(tienda4.shape)

tienda.head()

(2359, 12)
(2359, 12)
(2359, 12)
(2358, 12)


Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   8   4.60971 -74.08175  
1  Tarjeta de crédito                   4   6.25184 -75.56359  
2  Tarjeta de crédito                   1  10.39972 -75.51444  
3               Nequi                   1   3.43722 -76.52250  
4               Nequi                   1   6.25184 -75.56359

In [4]:
# Exploración de los DataFrames
import pandas as pd

# Mostrar las primeras 5 filas de cada tienda
print("********* Tiendas *********")

print("Tienda 1 - Primeras filas:")
print(tienda.head())
print("\nTienda 2 - Primeras filas:")
print(tienda2.head())
print("\nTienda 3 - Primeras filas:")
print(tienda3.head())
print("\nTienda 4 - Primeras filas:")
print(tienda4.head())

# Información de columnas y tipos de datos

print("********* Informacion de las tiendas *********")

print("\nInformación de Tienda 1:")
print(tienda.info())
print("\nInformación de Tienda 2:")
print(tienda2.info())
print("\nInformación de Tienda 3:")
print(tienda3.info())
print("\nInformación de Tienda 4:")
print(tienda4.info())

# Resumen estadístico de columnas numéricas

print("********* Resumen estadistico por tienda *********")

print("\nResumen estadístico de Tienda 1:")
print(tienda.describe())
print("\nResumen estadístico de Tienda 2:")
print(tienda2.describe())
print("\nResumen estadístico de Tienda 3:")
print(tienda3.describe())
print("\nResumen estadístico de Tienda 4:")
print(tienda4.describe())

# Verificar valores nulos
print("********* Valores nulos *********")

print("\nValores nulos en Tienda 1:")
print(tienda.isnull().sum())
print("\nValores nulos en Tienda 2:")
print(tienda2.isnull().sum())
print("\nValores nulos en Tienda 3:")
print(tienda3.isnull().sum())
print("\nValores nulos en Tienda 4:")
print(tienda4.isnull().sum())

# Listar columnas
print("\nColumnas de Tienda 1:")
print(tienda.columns)

********* Tiendas *********
Tienda 1 - Primeras filas:
            Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito            

## Exploración de Datos

### Estructura de los Datos
- **Columnas**: Producto, Categoría del Producto, Precio, Costo de envío, Fecha de Compra, Vendedor, Lugar de Compra, Calificación, Método de pago, Cantidad de cuotas, lat, lon.
- **Número de filas**:
  - Tienda 1: [reemplaza con el resultado de tienda.shape[0]]
  - Tienda 2: [reemplaza con el resultado de tienda2.shape[0]]
  - Tienda 3: [reemplaza con el resultado de tienda3.shape[0]]
  - Tienda 4: [reemplaza con el resultado de tienda4.shape[0]]
- **Tipos de datos**: Precio y Costo de envío (float64), Calificación y Cantidad de cuotas (int64), Fecha de Compra y otras categóricas (object), lat y lon (float64).

### Observaciones
- **Valores nulos**: [Indica si hay nulos o no, por ejemplo, "No se encontraron valores nulos en ninguna tienda"].
- **Rangos**:
  - Precios: [Mínimo y máximo observados, por ejemplo, "entre 7,600 y 2,977,000 en Tienda 1"].
  - Costos de envío: [Mínimo y máximo].
  - Calificaciones: Entre 1 y 5.
- **Categorías**: Incluyen Electrónicos, Muebles, Juguetes, Electrodomésticos, [otras categorías si las ves].
- **Métodos de pago**: Tarjeta de crédito, Nequi, [otros si los ves].
- **Patrones**: [Por ejemplo, "Tienda 1 tiene productos caros como Microondas, Tienda 2 parece tener más variedad"].

### Notas para el análisis
- Usar `Precio` para calcular ingresos totales.
- Agrupar por `Categoría del Producto` para ventas por categoría.
- Usar `Calificación` para promedio de valoraciones.
- Contar `Producto` para productos más/menos vendidos.
- Usar `Costo de envío` para promedio por tienda.

#1. Análisis de facturación



In [5]:
# 1. Análisis de facturación
ingreso_tienda1 = tienda['Precio'].sum()
ingreso_tienda2 = tienda2['Precio'].sum()
ingreso_tienda3 = tienda3['Precio'].sum()
ingreso_tienda4 = tienda4['Precio'].sum()

ingresos_totales = pd.DataFrame({
    'Tienda': ['Tienda 1', 'Tienda 2', 'Tienda 3', 'Tienda 4'],
    'Ingreso Total': [ingreso_tienda1, ingreso_tienda2, ingreso_tienda3, ingreso_tienda4]
})

print("Ingresos totales por tienda:")
print(ingresos_totales)

ingresos_totales.to_csv('ingresos_totales.csv', index=False)

Ingresos totales por tienda:
     Tienda  Ingreso Total
0  Tienda 1   1.150880e+09
1  Tienda 2   1.116344e+09
2  Tienda 3   1.098020e+09
3  Tienda 4   1.038376e+09


Los ingresos totales de cada tienda muestran que Tienda 1 tiene el mayor ingreso con 1.150 millones, seguida de Tienda 2 con 1.116 millones. Las Tiendas 3 y 4 presentan ingresos similares, alrededor de 1.098 millones y 1.038 millones, respectivamente. Esto indica que, en términos generales, la Tienda 1 tiene un rendimiento más alto, aunque la diferencia no es drástica, lo que podría significar que todas las tiendas están manteniendo un nivel de ventas competitivo.

# 2. Ventas por categoría

In [6]:
# 2. Ventas por categoría
ventas_categoria_tienda1 = tienda.groupby('Categoría del Producto')['Precio'].sum().reset_index()
ventas_categoria_tienda2 = tienda2.groupby('Categoría del Producto')['Precio'].sum().reset_index()
ventas_categoria_tienda3 = tienda3.groupby('Categoría del Producto')['Precio'].sum().reset_index()
ventas_categoria_tienda4 = tienda4.groupby('Categoría del Producto')['Precio'].sum().reset_index()

ventas_categoria_tienda1.columns = ['Categoría', 'Ingreso Tienda 1']
ventas_categoria_tienda2.columns = ['Categoría', 'Ingreso Tienda 2']
ventas_categoria_tienda3.columns = ['Categoría', 'Ingreso Tienda 3']
ventas_categoria_tienda4.columns = ['Categoría', 'Ingreso Tienda 4']

ventas_categorias = ventas_categoria_tienda1.merge(
    ventas_categoria_tienda2, on='Categoría', how='outer'
).merge(
    ventas_categoria_tienda3, on='Categoría', how='outer'
).merge(
    ventas_categoria_tienda4, on='Categoría', how='outer'
).fillna(0)

print("Ventas por categoría para cada tienda:")
print(ventas_categorias)

ventas_categorias.to_csv('ventas_categorias.csv', index=False)

Ventas por categoría para cada tienda:
                 Categoría  Ingreso Tienda 1  Ingreso Tienda 2  \
0  Artículos para el hogar        12698400.0        14746900.0   
1     Deportes y diversión        39290000.0        34744500.0   
2        Electrodomésticos       363685200.0       348567800.0   
3             Electrónicos       429493500.0       410831100.0   
4   Instrumentos musicales        91299000.0       104990300.0   
5                 Juguetes        17995700.0        15945400.0   
6                   Libros         8784900.0        10091200.0   
7                  Muebles       187633700.0       176426300.0   

   Ingreso Tienda 3  Ingreso Tienda 4  
0        15060000.0        15074500.0  
1        35593100.0        33350100.0  
2       329237900.0       283260200.0  
3       410775800.0       409476100.0  
4        77380900.0        75102400.0  
5        19401100.0        20262200.0  
6         9498700.0         9321300.0  
7       201072100.0       192528900.0  


El análisis de ventas por categoría revela que Electrodomésticos y Electrónicos son las categorías más fuertes en términos de ingresos, especialmente para Tienda 1 y Tienda 2, con ventas significativas en ambas. Sin embargo, Instrumentos musicales y Juguetes tienen ingresos más bajos, sugiriendo que estas categorías podrían necesitar más promoción o una oferta más atractiva para incrementar su rendimiento. Las diferencias en ingresos por tienda en categorías específicas también pueden reflejar variaciones en el mercado y las preferencias regionales.

# 3. Calificación promedio de la tienda


In [7]:
# 3. Calificación promedio de la tienda
calificacion_tienda1 = tienda['Calificación'].mean()
calificacion_tienda2 = tienda2['Calificación'].mean()
calificacion_tienda3 = tienda3['Calificación'].mean()
calificacion_tienda4 = tienda4['Calificación'].mean()

calificaciones_promedio = pd.DataFrame({
    'Tienda': ['Tienda 1', 'Tienda 2', 'Tienda 3', 'Tienda 4'],
    'Calificación Promedio': [calificacion_tienda1, calificacion_tienda2, calificacion_tienda3, calificacion_tienda4]
})

print("Calificación promedio por tienda:")
print(calificaciones_promedio)

Calificación promedio por tienda:
     Tienda  Calificación Promedio
0  Tienda 1               3.976685
1  Tienda 2               4.037304
2  Tienda 3               4.048326
3  Tienda 4               3.995759


Las calificaciones promedio de las tiendas están bastante equilibradas, con Tienda 3 teniendo la calificación más alta (4.05), seguida por Tienda 2 (4.04). Esto sugiere que los clientes están generalmente satisfechos con las compras y experiencias en estas tiendas. Sin embargo, Tienda 1 y Tienda 4 tienen calificaciones ligeramente más bajas, lo que podría implicar áreas de mejora en la atención al cliente o en la experiencia general de compra.

# 4. Productos más y menos vendidos

In [9]:
# 4. Productos más y menos vendidos
ventas_productos_tienda1 = tienda['Producto'].value_counts().reset_index()
ventas_productos_tienda2 = tienda2['Producto'].value_counts().reset_index()
ventas_productos_tienda3 = tienda3['Producto'].value_counts().reset_index()
ventas_productos_tienda4 = tienda4['Producto'].value_counts().reset_index()

ventas_productos_tienda1.columns = ['Producto', 'Ventas Tienda 1']
ventas_productos_tienda2.columns = ['Producto', 'Ventas Tienda 2']
ventas_productos_tienda3.columns = ['Producto', 'Ventas Tienda 3']
ventas_productos_tienda4.columns = ['Producto', 'Ventas Tienda 4']

mas_vendidos_tienda1 = ventas_productos_tienda1.head(1)
menos_vendidos_tienda1 = ventas_productos_tienda1.tail(1)
mas_vendidos_tienda2 = ventas_productos_tienda2.head(1)
menos_vendidos_tienda2 = ventas_productos_tienda2.tail(1)
mas_vendidos_tienda3 = ventas_productos_tienda3.head(1)
menos_vendidos_tienda3 = ventas_productos_tienda3.tail(1)
mas_vendidos_tienda4 = ventas_productos_tienda4.head(1)
menos_vendidos_tienda4 = ventas_productos_tienda4.tail(1)

productos_extremos = pd.concat([
    mas_vendidos_tienda1.assign(Tienda='Tienda 1', Tipo='Más Vendido'),
    menos_vendidos_tienda1.assign(Tienda='Tienda 1', Tipo='Menos Vendido'),
    mas_vendidos_tienda2.assign(Tienda='Tienda 2', Tipo='Más Vendido'),
    menos_vendidos_tienda2.assign(Tienda='Tienda 2', Tipo='Menos Vendido'),
    mas_vendidos_tienda3.assign(Tienda='Tienda 3', Tipo='Más Vendido'),
    menos_vendidos_tienda3.assign(Tienda='Tienda 3', Tipo='Menos Vendido'),
    mas_vendidos_tienda4.assign(Tienda='Tienda 4', Tipo='Más Vendido'),
    menos_vendidos_tienda4.assign(Tienda='Tienda 4', Tipo='Menos Vendido')
])

print("Productos más y menos vendidos por tienda:")
print(productos_extremos)

Productos más y menos vendidos por tienda:
                     Producto  Ventas Tienda 1    Tienda           Tipo  \
0                  Microondas             60.0  Tienda 1    Más Vendido   
50               Celular ABXY             33.0  Tienda 1  Menos Vendido   
0   Iniciando en programación              NaN  Tienda 2    Más Vendido   
50              Juego de mesa              NaN  Tienda 2  Menos Vendido   
0               Kit de bancas              NaN  Tienda 3    Más Vendido   
50    Bloques de construcción              NaN  Tienda 3  Menos Vendido   
0                    Cama box              NaN  Tienda 4    Más Vendido   
50         Guitarra eléctrica              NaN  Tienda 4  Menos Vendido   

    Ventas Tienda 2  Ventas Tienda 3  Ventas Tienda 4  
0               NaN              NaN              NaN  
50              NaN              NaN              NaN  
0              65.0              NaN              NaN  
50             32.0              NaN              NaN  
0

El análisis de los productos más y menos vendidos destaca algunos productos clave que impulsan las ventas. Tienda 1 tiene el Microondas como su producto más vendido, lo que refleja una fuerte demanda de electrodomésticos, mientras que el Celular ABXY es el menos vendido. Tienda 2, Tienda 3 y Tienda 4 presentan productos con un rendimiento más variado, pero no hay un patrón claro debido a la falta de datos en algunos productos. Este análisis sugiere que las tiendas pueden beneficiarse de una mayor estrategia de promoción o ajuste en los productos menos vendidos, como el Juego de mesa y el Bloques de construcción.




# 5. Envío promedio por tienda

In [11]:
# Promedio de costos de envío por tienda
envio_tienda1 = tienda['Costo de envío'].mean()
envio_tienda2 = tienda2['Costo de envío'].mean()
envio_tienda3 = tienda3['Costo de envío'].mean()
envio_tienda4 = tienda4['Costo de envío'].mean()

# Mostrar resultados
print("Promedio de costo de envío por tienda:")
print(f"Tienda 1: {envio_tienda1}")
print(f"Tienda 2: {envio_tienda2}")
print(f"Tienda 3: {envio_tienda3}")
print(f"Tienda 4: {envio_tienda4}")


Promedio de costo de envío por tienda:
Tienda 1: 26018.60958033065
Tienda 2: 25216.235693090293
Tienda 3: 24805.680373039424
Tienda 4: 23459.457167090754


En términos de costos de envío, Tienda 1 tiene el costo promedio más alto (26,018), seguido de cerca por Tienda 2 (25,216). Esto podría implicar que algunas tiendas están incurriendo en mayores costos logísticos que otras. Las tiendas deberían explorar oportunidades para optimizar sus procesos de envío y reducir estos costos, lo cual podría mejorar su rentabilidad. Tienda 3 y Tienda 4 tienen costos de envío más bajos, lo que sugiere que están manejando sus operaciones logísticas de manera más eficiente.

### Resumen de Análisis

1. **Análisis de Facturación:**
   - La **Tienda 1** presenta el mayor ingreso total, seguida por la **Tienda 2**, mientras que las **Tiendas 3** y **Tienda 4** tienen ingresos más bajos, aunque la diferencia no es tan significativa, lo que sugiere un rendimiento equilibrado en términos de ventas. Sin embargo, la **Tienda 1** lidera en ingresos, mientras que la **Tienda 4** tiene el rendimiento más bajo en este aspecto.

2. **Ventas por Categoría:**
   - Las categorías de **Electrodomésticos** y **Electrónicos** generan más ventas, especialmente en **Tienda 1** y **Tienda 2**, mientras que categorías como **Instrumentos musicales** y **Juguetes** tienen un desempeño más bajo. Esto indica que las tiendas podrían enfocarse en mejorar las categorías con menores ventas y aumentar su promoción para diversificar las fuentes de ingreso.

3. **Calificación Promedio:**
   - Las calificaciones promedio de las tiendas son positivas, siendo la **Tienda 3** la que obtiene la calificación más alta (4.05), y la **Tienda 1** la más baja (3.98). Aunque las calificaciones son bastante buenas en general, la **Tienda 1** podría mejorar en términos de atención al cliente para elevar su puntaje.

4. **Productos Más y Menos Vendidos:**
   - En **Tienda 1**, el **Microondas** es el producto más vendido, mientras que el **Celular ABXY** es el menos vendido. En las otras tiendas, los productos más y menos vendidos varían más. Esto sugiere que las tiendas deberían centrarse en mejorar la visibilidad de los productos menos vendidos, ajustando su oferta y promoción de acuerdo con la demanda de los consumidores.

5. **Envío Promedio por Tienda:**
   - La **Tienda 1** tiene el costo de envío más alto, mientras que **Tienda 4** tiene el más bajo, lo que puede influir en los márgenes de ganancia. Las tiendas con menores costos de envío, como **Tienda 4**, tienen una ventaja en este aspecto, y la **Tienda 1** podría optimizar sus costos de envío para mejorar su rentabilidad.

---

### Recomendación Final

**¿Cuál tienda debería vender el Sr. Juan?**

Basado en el análisis de los ingresos, ventas, calificaciones y costos de envío, la **Tienda 4** debería ser la que el Sr. Juan considere vender. A pesar de tener un costo de envío ligeramente más bajo que otras tiendas, su rendimiento en ventas y calificación promedio es inferior al de las demás. Al vender esta tienda, el Sr. Juan podría mejorar el rendimiento y focalizar esfuerzos en aumentar las ventas y la satisfacción del cliente en este negocio.

